text/0000-refresh-method-for-router-service.md

```diff
@@ -0,0 +1,47 @@
+- Start Date: 2020-05-23
+- Relevant Team(s): Ember.js
+- RFC PR: (after opening the RFC PR, update this with a link to it and update the file name)
+- Tracking: (leave this empty)
+
+# RouterService#refresh
+
+## Summary
+
+> Add a refresh method to the router service that calls refresh on the route specified by `currentRouteName`.
+
+## Motivation
+
+> We want to be able to call refresh the route specified by `currentRouteName` without relying on the send api, which we would like to deprecate. This enables us to get the latest data from the model hook.
```

Comment:
I think there are a number of better motivations. Deprecating `.send` isn't _really_ a motivation for adding this feature IMO (it's a nice side benefit though 😄). A better motivation (that I've seen in the wild a few places) is wanting to be able to trigger a route refresh from a centralized service, or to trigger a refresh from within a component (e.g. a "refresh button").

Reply:
Should also not mention `currentRouteName`, as the router service would be refreshing the models for **all** currently active routes not just the one referenced by `currentRouteName`.

Reply:
> all currently active routes

Just to make sure... "all active" means `currentRoute` and all its ancestors, right? Does this have a name? Maybe `currentRouteTree` or something could represent this in a data structure that makes sense?

Reply:
Yes, but I've decided to use "all currently active routes" because "active routes" does have a technical definition in the api.

Reply:
> Just to make sure... "all active" means currentRoute and all its ancestors, right? 

Ya, exactly. I don’t know that it has a name other than “currently active routes” or some variant thereof.